sim/common/cata/enchant_effects.go

```diff
@@ -380,6 +380,63 @@ func init() {
 		character.ItemSwap.RegisterOnSwapItemForEnchantEffect(4115, aura)
 	})
 
+	// Enchant: 4116, Spell: 75175 - Darkglow Embroidery
+	core.NewEnchantEffect(4116, func(agent core.Agent) {
+		character := agent.GetCharacter()
+
+		statAura := character.NewTemporaryStatsAura(
+			"Darkglow Embroidery Cata",
+			core.ActionID{SpellID: 75175},
+			stats.Stats{stats.Spirit: 480},
+			time.Second*15,
+		)
+
+		aura := core.MakeProcTriggerAura(&character.Unit, core.ProcTrigger{
+			Name:       "Darkglow Embroidery Cata",
+			ActionID:   core.ActionID{SpellID: 75171},
+			Callback:   core.CallbackOnSpellHitDealt | core.CallbackOnPeriodicDamageDealt | core.CallbackOnHealDealt,
+			ProcMask:   core.ProcMaskSpellDamage | core.ProcMaskSpellHealing,
+			Outcome:    core.OutcomeLanded,
+			ICD:        time.Second * 60,
+			ProcChance: 0.35,
+			Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+				statAura.Activate(sim)
+			},
+		})
+
+		character.ItemSwap.RegisterOnSwapItemForEnchantEffect(4115, aura)
+	})
+
+	// Enchant: 4118, Spell: 75178 - Swordguard Embroidery
+	core.NewEnchantEffect(4118, func(agent core.Agent) {
+		character := agent.GetCharacter()
+
+		procAura := character.NewTemporaryStatsAura("Swordguard Embroidery Cata", core.ActionID{SpellID: 75178}, stats.Stats{stats.AttackPower: 1000, stats.RangedAttackPower: 1000}, time.Second*15)
+		icd := core.Cooldown{
+			Timer:    character.NewTimer(),
+			Duration: time.Second * 55,
+		}
+		procAura.Icd = &icd
+
+		character.GetOrRegisterAura(core.Aura{
+			Label:    "Swordguard Embroidery",
+			Duration: core.NeverExpires,
+			OnReset: func(aura *core.Aura, sim *core.Simulation) {
+				aura.Activate(sim)
+			},
+			OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+				if !result.Landed() || !spell.ProcMask.Matches(core.ProcMaskMeleeOrRanged) {
+					return
+				}
+
+				if icd.IsReady(sim) && sim.RandomFloat("Swordguard") < 0.2 {
```

Comment:
15% chance not 20% https://www.wowhead.com/cata/spell=75177/swordguard-embroidery

Reply:
pushed changes